nixos/modules/tasks/filesystems/overlayfs.nix

```diff
@@ -0,0 +1,151 @@
+{ config, lib, pkgs, utils, ... }:
+
+let
+  toplevelConfig = config;
+
+  # Returns a service that creates the required directories before the mount is
+  # created.
+  preMountService = _name: fs:
+    let
+      rwName = "rw-${utils.escapeSystemdPath fs.mountPoint}";
+
+      # The scripted initrd contains some magic to add the prefix to the
+      # paths just in time, so we don't add it here.
+      prefix =
+        if config.boot.initrd.systemd.enable && fs.neededForBoot then
```

Comment:
This should be `utils.fsNeededForBoot fs`, because it has implicit checks for things like `/etc`. (Arguably, `fs.neededForBoot` should have that logic included by default but that's not really the point :P)